src/views/docs-view/types.ts

```diff
@@ -40,6 +42,21 @@ export interface DocsViewProps {
 	 * An optional, human-readable name to be rendered where the section is referenced as a versioned project
 	 */
 	projectName?: string
+
+	/**
+	 * Product data. Note we include `currentRootDocsPath` for version switching.
+	 */
+	product: ProductWithCurrentRootDocsPath
```

Comment:
Meant to type in the review submission -- this is my only blocking comment!

Reply:
I can see where you're coming from here - `DocsView` doesn't use the `product` prop, so in a sense it's not accurate to say that it's part of the props interface of `DocsView`.

The rationale is for why I felt it was worth including is:

1. It's very helpful to have our `getStaticProps` return typed. For that case, we do need the `product` prop to be documented. We could create a separate type, but...
2. Anything returned by `getStaticProps` will be available to the view component, so it feels helpful to know what's available. In this case for example, knowing that `product` data is available, we could likely simplify the implementation of `DocsView` by grabbing `product` and using it directly rather than having to call the `useCurrentProduct()` hook at render time. Whether or not we do this (this PR makes no changes so we don't do it), the `product` prop will still be returned from `getStaticProps` and available to `DocsView`, so it feels accurate to me to have it documented in `DocsViewProps`.

In general, it feels helpful to have `getStaticProps` and the view it will pass props to share types. So despite the fact that `DocsView` doesn't use the incoming `product` prop directly, to me it feels worth the inaccuracy in that sense in order to accurately reflect the relationship between the return type of `getStaticProps` and the incoming prop types of `DocsView`.

Happy to reconsider though if there are strong opinions here - would love to hear other thoughts cc @kendallstrautman @BRKalow

Reply:
No argument with improving the TypeScript and the return types included for `getStaticProps`! I'd just like the block comment above the `product` property to be updated for accuracy. 😊

Reply:
Ah cool, I'll update the comment! One sec...